File: console-tools/christmas tree/Tree.java
/**
 * Prints a tree with trunk on stdout
 *
 * @author pkwien93
 * @version 2008-10-14
 */
public class Tree {
	public static void main(String[] args) {
		int trunkheight = 2;
		int height = 7;

		tree(height);
		trunk(trunkheight, height);
	}

	/**
	 * prints a tree on stdout
	 * @param height number of rows
	 */
	public static void tree(int height) {
		//loop to print the rows of the tree
		for (int row = 0; row < height; row++) {
			for (int col = 0; col < (height * 2 - 1); col++) {
				if (col >= (height - row - 1) && col <= (height + row - 1))
					System.out.print("*");
				else
					System.out.print(" ");
			}
			System.out.println();
		}
	}
	/**
	 * prints the trunk of an tree on stdout
	 * @param trunkheight number of rows for the trunk
	 * @param height number of rows for the tree (neccessary to center the trunk)
	 */
	public static void trunk(int trunkheight, int height) {
		//loop to print the rows of the trunk
		for (int trunk = 0; trunk < trunkheight; trunk++) {
			//loop for the spaces to center the trunk
			for (int i = 0; i < height - 2; i++) {
				System.out.print(" ");
			}
			//printing the trunk
			System.out.println("***");
		}
	}
}

File: console-tools/christmas tree/tree.c
#include <stdio.h>

int main(int argc, char *argv[]) {
	/* defines the height of the tree (number of rows) */
	int height = 7;
	/* defines the trunk of the tree (number of rows) */
	int trunkheight = 2;

	/* use given program parameter for height and trunkheight if present */
	if(argc == 3) {
		height = atoi(argv[1]);
		trunkheight = atoi(argv[2]);
	}

	//loop variables
	int c = 0, r = 0;
	
	/** print tree */
	for(r = 0; r < height; r++) {
		for(c = 0; c < (height * 2 - 1); c++) {
			if(c >= (height - r - 1) && c <= (height + r -1))
				printf("*");
			else
				printf(" ");
		}
		printf("\n");
	}

	/** print trunk */
	for(r = 0; r < trunkheight; r++) {
		for(c = 0; c < height - 2; c++) {
			printf(" ");
		}
		printf("***\n");
	}
};